src/server/mod.rs

```diff
@@ -308,6 +308,10 @@ impl ServerConfig {
     /// Sets a single certificate chain and matching private key.  This
     /// certificate and key is used for all subsequent connections,
     /// irrespective of things like SNI hostname.
+    /// ***Note that the end-entity certificate must have the 
+    /// [Subject Alternative Name](https://en.wikipedia.org/wiki/Subject_Alternative_Name) 
+    /// extension to describe, e.g., the valid DNS name. The `commonName` field is 
+    /// disregarded.***
```

Comment:
Please add a reference to [RFC 6125 Section 4.1](https://tools.ietf.org/html/rfc6125#section-4.1).